### ISSUE TITLE
Deliverable

### ISSUE THREAD

**Issue by pendo324** [PuppetCase/Assignment2#2]
We only have to submit one per group, right? I've got the kernel stuff up with a patch file and report. Has anyone worked on the report for part 2? We need to submit it today, I think.

**Comment by pjime032** [comment]
oh crap!!! i totally forgot!! let me get on that

**Comment by alaingalvan** [comment]
I get out of work at 7, are you guys on campus, I can help! ;_;

**Comment by pjime032** [comment]
nah. i'm at home. just finished the report for part 2. can you compile it
together?

**Comment by alaingalvan** [comment]
Sure thing!

**Comment by pjime032** [comment]
okay, i've uplodaed to git ny part of the report and the README file.

**Comment by pjime032** [comment]
or not... damn forgot to sync. let me do that.

**Comment by pjime032** [comment]
okay, done.

**Comment by alaingalvan** [comment]
Running through everything now!

**Comment by alaingalvan** [comment]
Almost done!

**Comment by alaingalvan** [comment]
Guys, this part of the requirements is making me think submissions are still individual:

> The submission should include only your source code and report.

**Comment by pendo324** [comment]
I think that just means that they don't want the actual compiled program. 

Also, there's this part
> 8) Clearly state your group members and who does what in your report.

**Comment by pjime032** [comment]
Right. No compiled files.

**Comment by alaingalvan** [comment]
> Also, there's this part

It's at the end of the file. ;D

**Comment by alaingalvan** [comment]
So I'm confused with the Policies, I'm reading that:

> You must work in a group of two people on this exercise. We encourage high-level discussions among the groups to help each other understand the concepts and principles. However, code-level discussion is prohibited. We will use anti-plagiarism tools to detect violations of this policy.

Does this mean that our submissions must be our own implementations? 

## Part 1 deliverables: 
(1) A README file, which describes how we can compile and run your code;
(2) Your source code, must include a Makefile;
(3) A patch file that represents the changes that you made to the vanilla 2.6.36 kernel to implement the new system call. It is okay to include the other system call that you added in Lab 1;
(4) Your report, which discusses the output of your program without Pthread synchronization and the one with Pthread synchronization, as well as the reason for the difference. In addition, the output of 
dmesg that shows the process and thread information.

## Part 2 Deliverables
Part 2 deliverables: 
(1) A README file, which describes how we can compile and run your code;
(2) Your source code, must include a Makefile;
(3) Your  report,  which  discusses  the  design  of  your  News  Conference  program  and  how  Pthread synchronization is used in your program.

## Overall
1)  The submission should include deliverables for both Part 1 and Part 2. Submit a .zip file that contains all files; put Part 1 and Part 2 in separate directories.
2)  The submission should include only your source code and report. Do not submit your binary code.
3)  Your code must be able to compile; otherwise, you will receive a grade of zero.
4)  Your code should not produce anything else other than the required information in the output file.
5)  Your code must validate command line parameters to make sure that only numbers can be accepted.
6)  If you code is partially completed, also explain in the report what has been completed and the status of the missing parts.
7)  Provide sufficient comments in your code to help the TA understand your code. This is important for you to get at least partial credit in case your submitted code does not work properly.
8) Clearly state your group members and who does what in your report.

**Comment by pendo324** [comment]
I think that might mean like talking across different groups, not sure

**Comment by pjime032** [comment]
yeah... intra-group code sharing is bad... inter-group sharing is good.
PuppetCase
